lib/internal/modules/cjs/loader.js

```diff
@@ -753,7 +756,7 @@ Module.prototype._compile = function(content, filename) {
       undefined,
       false,
       undefined,
-      [],
+      [processBufferContext],
```

Comment:
note that this will break referring to buffer and console via `global`, so im not really sure it could ever be safe to merge

Reply:
If there are any cases of this it is a very clear deprecation warning. Note the actual deprecation can be in many many majors. The important part is getting the warning going for ES modules.

Reply:
I mean with the current changes global.Buffer and Buffer might not be equal. that shouldn't be observable.

Reply:
Good point, I can add the code to ensure these are synchronized.

Reply:
Synchronization code included.